bluebox/backends/backend_pyaudio.py

```diff
@@ -0,0 +1,114 @@
+"""backend_pyaudio.py
+
+This file contains the PyAudio backend for bluebox.
+"""
+
+import typing as t
+import logging
+import pyaudio
+from .base import BlueboxBackend
+
+
+class PyAudioBackend(BlueboxBackend):
+    """PyAudioBackend class for the PyAudio backend."""
+
+    _stream: pyaudio.Stream
+    _stream_open: bool = False
+    _device: t.Union[int, None]
+
+    def __init__(
+                self,
+                sample_rate: float = 44100.0,
+                channels: int = 1,
+                amplitude: float = 1.0,
+                logger: t.Optional[logging.Logger] = None,
+                device: t.Union[int, None] = None) -> None:
+        """Initialize the PyAudio backend."""
+        super().__init__(sample_rate, channels, amplitude, logger)
+        self._device = device
+
+    def _get_stream(self) -> pyaudio.Stream:
+        """Get the PyAudio stream."""
+        if not self._stream_open:
+            self._stream = pyaudio.PyAudio().open(
+                format=pyaudio.paFloat32,
+                channels=self._ch,
+                rate=int(self._sr),
+                output=True,
+                output_device_index=self._device)
+            self._stream_open = True
+        return self._stream
+
+    def play(self, data: t.MutableSequence[float]) -> None:
+        """Play the given data."""
+        self._get_stream().write(data)  # type: ignore
+
+    def play_all(self, queue: t.Iterable[t.MutableSequence[float]]) -> None:
+        """Play all the items until the end."""
+        for data in queue:
+            self.play(data)
+
+    def stop(self) -> None:
+        """Stop playing the data."""
+        if self._stream_open:
+            self._stream.stop_stream()
+
+    def close(self) -> None:
+        """Close the backend."""
+        if self._stream_open:
+            self._stream.close()
+            self._stream_open = False
+
+    def __del__(self) -> None:
+        """Close the backend."""
+        self.close()
+
+
+class PyAudioBackendNonBlocking(PyAudioBackend):
+    """PyAudioBackendNonBlocking class for the PyAudio backend."""
+
+    def _get_stream(self, callback: t.Callable) -> pyaudio.Stream:
+        """Get the PyAudio stream."""
```

Comment:
add arg docs, also keep params descriptive (channels not ch) (also be consistent)...the privates can still be _sr, _ch etc.